packages/autocomplete-plugin-algolia-insights/src/createAlgoliaInsightsPlugin.ts

```diff
@@ -196,16 +201,45 @@ export function createAlgoliaInsightsPlugin(
             insights,
           },
         });
+
+        if (preserveUserToken) {
+          preservedUserToken = userToken;
+        }
```

Comment:
I'm realizing that while we allow passing `userToken` in params, we don't commit it to `insightsClient` when passed this way.

If we did, you could avoid this local `preserveUserToken` variable and get it back using `insightsClient('getUserToken')` wdyt?

Reply:
> I'm realizing that while we allow passing userToken in params, we don't commit it to insightsClient when passed this way.

If it's passed as init params, it will be retrieved using `getUserToken` and updated in the context. Was it what you had in mind?

> If we did, you could avoid this local preserveUserToken variable and get it back using insightsClient('getUserToken') wdyt?

Good idea, I'll switch to it.

Reply:
> If it's passed as init params, it will be retrieved using getUserToken and updated in the context. Was it what you had in mind?

No, I meant to also commit it to `insightsClient` by calling `insightsClient('setUserToken')` internally so even if the `userToken` is passed directly to the plugin, it's also reflected in the Insights client (and allows you to retrieve it from it later on).

Reply:
I see. I believe the only ways for a user to set a user token from the plugin are covered:
- with `insightsInitParams`, which are passed to the init method of the insights client
- with the functional interface API, which calls `setUserToken` on the insights client

Reply:
Yes I just saw it was indeed! Scratch this then :)